EventFilter/L1TRawToDigi/src/implementations_stage2/EtSumPacker.cc

```diff
@@ -36,30 +36,42 @@ namespace stage2 {
       edm::Handle<EtSumBxCollection> etSums;
       event.getByToken(static_cast<const CommonTokens*>(toks)->getEtSumToken(), etSums);
 
-      uint32_t et_word = 0;
-      uint32_t ht_word = 0;
-      uint32_t met_word = 0;
-      uint32_t mht_word = 0;
-
+      std::vector<uint32_t> load;
+      int nBx = 0;
       for (int i = etSums->getFirstBX(); i <= etSums->getLastBX(); ++i) {
+
+	 uint32_t et_word = 0;
+	 uint32_t ht_word = 0;
+	 uint32_t met_word = 0;
+	 uint32_t mht_word = 0;
+
          for (auto j = etSums->begin(i); j != etSums->end(i); ++j) {
 	   uint32_t word = std::min(j->hwPt(), 0xFFF);
 	   if ((j->getType()==l1t::EtSum::kMissingEt) || (j->getType()==l1t::EtSum::kMissingHt))
 	     word = word | ((j->hwPhi() & 0xFF) << 12);
 	   
-	   if (j->getType()==l1t::EtSum::kTotalEt)   et_word = word;
-	   if (j->getType()==l1t::EtSum::kTotalHt)   ht_word = word;
-	   if (j->getType()==l1t::EtSum::kMissingEt) met_word = word;
-	   if (j->getType()==l1t::EtSum::kMissingHt) mht_word = word;
+	   if (j->getType()==l1t::EtSum::kTotalEt)       et_word  |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFP0)   et_word  |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kTotalHt)       ht_word  |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFM0)   ht_word  |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kMissingEt)     met_word |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFP1)   met_word |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kMissingHt)     mht_word |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFM1)   mht_word |= (word << 28);
```

Comment:
Hi Brian, Vladmir,

I’ve updated the EtSumUnpacker to include the min bias bits and also the ETTEM and 
ETMHF, HTMHF.

Brian, if you could check what I’ve done seems correct that would be great.

The PR is here:

https://github.com/cms-l1t-offline/cmssw/pull/371 <https://github.com/cms-l1t-offline/cmssw/pull/371>

Cheers,
Aaron



> On 30 Jun 2016, at 17:21, Aaron Bundock <aaron.bundock@cern.ch> wrote:
> 
> Hi Brian,
> 
> I’ll make these changes and put in a PR later today.
> 
> Cheers,
> Aaron
> 
> 
>> On 30 Jun 2016, at 16:32, blwiner <notifications@github.com <mailto:notifications@github.com>> wrote:
>> 
>> In EventFilter/L1TRawToDigi/src/implementations_stage2/EtSumPacker.cc <https://github.com/cms-l1t-offline/cmssw/pull/333#discussion_r69142948>:
>> 
>> >           for (auto j = etSums->begin(i); j != etSums->end(i); ++j) {
>> >  	   uint32_t word = std::min(j->hwPt(), 0xFFF);
>> >  	   if ((j->getType()==l1t::EtSum::kMissingEt) || (j->getType()==l1t::EtSum::kMissingHt))
>> >  	     word = word | ((j->hwPhi() & 0xFF) << 12);
>> >  	   
>> > -	   if (j->getType()==l1t::EtSum::kTotalEt)   et_word = word;
>> > -	   if (j->getType()==l1t::EtSum::kTotalHt)   ht_word = word;
>> > -	   if (j->getType()==l1t::EtSum::kMissingEt) met_word = word;
>> > -	   if (j->getType()==l1t::EtSum::kMissingHt) mht_word = word;
>> > +	   if (j->getType()==l1t::EtSum::kTotalEt)       et_word  |= word;
>> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFP0)   et_word  |= (word << 28);
>> > +	   if (j->getType()==l1t::EtSum::kTotalHt)       ht_word  |= word;
>> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFM0)   ht_word  |= (word << 28);
>> > +	   if (j->getType()==l1t::EtSum::kMissingEt)     met_word |= word;
>> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFP1)   met_word |= (word << 28);
>> > +	   if (j->getType()==l1t::EtSum::kMissingHt)     mht_word |= word;
>> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFM1)   mht_word |= (word << 28);
>> I agree these need to be changed. Perhaps @jimbrooke <https://github.com/jimbrooke> or @bundocka <https://github.com/bundocka> want to do this? I originally made these bug fixes so that I could test mutliboard uGT readout, but this is more their territory. If they want me to do it, I can; but I would feel more comfortable it they put in the changes. -Brian
>> … <x-msg://15/#>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub, or mute the thread.
>> 
>>  <http://etsumpacker.cc/> <https://github.com/cms-l1t-offline/cmssw/pull/333#discussion_r69123226> <https://github.com/cms-l1t-offline/cmssw/pull/333/files/9d205b3c737d05eaebd4b301c6fe7750570b2117#r69123226> <https://github.com/notifications/unsubscribe/AElaBv9issVRE23kdJzLPmPO1EHfk9b4ks5qQ7d9gaJpZM4IlnuI>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub <https://github.com/cms-l1t-offline/cmssw/pull/333/files/9d205b3c737d05eaebd4b301c6fe7750570b2117#r69142948>, or mute the thread <https://github.com/notifications/unsubscribe/AMqRT4LEYZm6RKLbOvcCbU4TEDVV581rks5qQ9NogaJpZM4IlnuI>.
>> 
>